### PR TITLE
Old style switches: changed the loiter switch to have priorit over ma…

### LINKED ISSUE #6733: Auto Pause: Total disaster.
Today I had a total crash with a novice while I was teaching him how to fly the PX4!

Why?? Just because he could not understand nor he could learn that in order to PAUSE his flight he has to 

FIRST: switch to MISSION
and then
SECOND: to switch Auto Pause

it is just overwhelming for the pilot to think of this combination, especially in dare situation when you most likely need Auto Pause. It is against all the rules in a real airplane cockpit as well.

It is crazzzzyyyy!

It is like having your phone ringing in a meeting and then in order to MUTE it you have to first hit home button and then hold volume down for 2 seconds.

Or it is like if you are in the car listening to music CD and have your phone ringing first hit on the Radio Button and then hit CD pause in order to Pause your CD.

Now I have to make an input mixer setting in the Horus in order to do it with just one button, given all the processing power I have in the Pixhawk. Is it hard to make it function like RTL?

This becomes the N1 reason for crashes of novice for the last 6 months I experience. Nobody wants to fly PX4 at the club already. And it is hard for me to persuade them to start using it again with such public crashes and disaster in front of all people and humiliation for me!

It is non sense, period.

Or maybe I'm stupid and I'm missing some hidden parameter that should allow this. If this is the case excuse me for the rant.

### PR DESCRIPTION
**Pull request by AndreasAntener**
…in switch

Fixes #6733 , I wanted to do this for a long time actually...

### CODE REVIEW

File: src/modules/commander/commander.cpp [hunk 1]
```diff
@@ -3400,6 +3400,18 @@ set_main_state_rc(struct vehicle_status_s *status_local)
 		/* if we get here mode was rejected, continue to evaluate the main system mode */
 	}
 
+	/* Loiter switch overrides main switch */
+	if (sp_man.loiter_switch == manual_control_setpoint_s::SWITCH_POS_ON) {
+		res = main_state_transition(status_local, commander_state_s::MAIN_STATE_AUTO_LOITER, main_state_prev, &status_flags, &internal_state);
+
+		if (res == TRANSITION_DENIED) {
+			print_reject_mode(status_local, "AUTO PAUSE");
```
Comment (LorenzMeier) [inline src/modules/commander/commander.cpp]: Can you change that to `HOLD`?

**Comment by julianoes** [comment]
@AndreasAntener thanks! This is workaround in the meantime, right? It just changes priorities but doesn't address that clunkyness of the nested setup?

**Comment by dagar** [comment]
Looks good. QGC update as well? I think the loiter/hold/pause naming is out of date there.

**Comment by AndreasAntener** [comment]
@julianoes I'm not sure about the clunkyness? Personally I still really like the nested setup, the missing HOLD priority was my only problem in general.

@dagar I'm not sure how QGC does it, is an update needed for this? If yes I can open an issue in general for the old style assignment.

**Comment by dagar** [comment]
The loiter naming is out of sync and the configuration gui might be expecting loiter to be nested.

My main objection is having two supported ways of doing things. It makes new work and maintenance work slightly harder. Things break on one side, but go unnoticed. If there's a viable path forward for making the new config work for you I think that's a better place to invest future effort.

**Comment by dagar** [comment]
I did a quick test and this seems to work as intended.

**Pull request merged**